main.go

```diff
@@ -83,6 +102,34 @@ func main() {
 	os.Exit(exitCode)
 }
 
+// Check status of SDS secrets from Istio to make sure SDS bootstrap was successful
+// This SDS data is populated at bootstrap Before the actual xDS connections are established so it should be safe to
+// just kill the envoy pod qith `quitquitquit` if we are in a bad state here
+func checkEnvoyIstioSDS(host string) error {
+	url := fmt.Sprintf("%s/certs", host)
+
+	rsp := typhon.NewRequest(context.Background(), "GET", url, nil).Send().Response()
+
+	certs := &Certificates{}
+
+	err := rsp.Decode(certs)
+	if err != nil {
+		return err
+	}
+
+	// Check number of CERTS is 1 or 0, if it is hit the KILL Endpoint
+	if len(certs.Certificates) < 2 {
+		// Proper logging! with timestamp
+		fmt.Println("envoy-preflight: Only 1 certificate found in :15000/certs, killing envoy pod\n")
+		_ = typhon.NewRequest(context.Background(), "POST", fmt.Sprintf("%s/quitquitquit", host), nil).Send().Response()
```

Comment:
This may block forever, unless there is some built in timeout. Would it be better to pass in a context with deadline?

Reply:
I just copied the other `typhon` request from the same code we are using now but i see the point.

I guess the upstream code rely on the kubernetes to trigger a restart of a pod that never comes up ( they even some comments about that )

I would be ok adding the timeout to this request, and i guess all others, but i am not sure how to do that exactly , i can see this example in the typhon library repo

https://github.com/monzo/typhon/blob/304194740fd4678b43c35c2b04652ecc0acc719f/examples/simple.go#L35
```
	c, cancel := context.WithTimeout(context.Background(), 10*time.Second)
	defer cancel()
```

Reply:
> I guess the upstream code rely on the kubernetes to trigger a restart of a pod that never comes up ( they even some comments about that )

Actually that is only true if we had a liveness probe so i think you are right we should have timeout on all the requests, not just the one i added, or there is a risk that a `forever blocking` request hangs the startup of a pod

Reply:
not that the code already expect to wait forever because it expects a liveness probe to deal with a blocked pod 

https://github.com/influxdata/envoy-preflight/blob/master/main.go#L140

Reply:
Good catch. The `context.Background()` should be fine in that case, but it could still benefit from a clarifying comment.

Reply:
added comment in https://github.com/influxdata/envoy-preflight/pull/5 

If we are good with this i will make a PR to idpe to use this fork so we can start run it on gateway workloads without enabling the new check flag which will do later on on stage and tools. Essentially that should be a noop

Reply:
I think https://github.com/influxdata/idpe/blob/051ac4e8e965988ed0b9faa7fca969476deb9677/scripts/build-cd-base.bash#L125 and https://github.com/influxdata/idpe/blob/051ac4e8e965988ed0b9faa7fca969476deb9677/tools.go#L14 are the only lines you will need to change in idpe to use our fork.

After that you will need to `go mod vendor && go mod tidy` iirc to satisfy CI.

Reply:
thanks for pointing those out @mark-rushakoff , i was also going to set a replace in `go.mod` , i guess i can just override the monzo one . do we need this ? 

```
git diff --cached go.mod
diff --git a/go.mod b/go.mod
index a6b09df57..0527fde27 100644
--- a/go.mod
+++ b/go.mod
@@ -184,6 +184,7 @@ replace (
        github.com/docker/docker => github.com/docker/engine v17.12.0-ce-rc1.0.20200309214505-aa6a9891b09c+incompatible
        github.com/golang/protobuf => github.com/golang/protobuf v1.3.3
        github.com/influxdata/platform => /dev/null
+       github.com/monzo/envoy-preflight => github.com/influxdata/envoy-preflight v0.0.0-20210325133213-8c837d7c36f1
        google.golang.org/genproto => google.golang.org/genproto v0.0.0-20200218151345-dad8c97a84f5
        istio.io/istio => github.com/influxdata/istio v0.0.0-20190828174436-ab0c162177a2
        k8s.io/api => k8s.io/api v0.0.0-20190708174958-539a33f6e817 // kubernetes-1.14.4
```

Reply:
After you change the bash script and tools.go references to use `influxdata` then the replace should not be necessary.